pkg/controller/storagecluster/cephobjectstores.go

```diff
@@ -107,7 +107,7 @@ func (r *ReconcileStorageCluster) newCephObjectStoreInstances(initData *ocsv1.St
 				Gateway: cephv1.GatewaySpec{
 					Port:      80,
 					Instances: 2,
-					Placement: defaults.DaemonPlacements["rgw"],
+					Placement: getPlacement(initData, "rgw"),
```

Comment:
I don't think this solves the problem.

The original issue was that `defaults.DaemonPlacements["rgw"]` doesn't include `NodeAffinity` which was moved out for some reason, and that still persists here.

When placement is not specified in StorageCluster, `getPlacement` falls back to `defaults.DaemonPlacements["rgw"]` and it still wouldn't get `NodeAffinity`. We need to inject NodeAffinity in that case.

Yes, the unit-tests pass because they were designed to pass. But it fails in most of the real world scenarios.

Reply:
`getPlacement()` sets the NodeAffinity: https://github.com/openshift/ocs-operator/blob/master/pkg/controller/storagecluster/placement.go#L27-L29

Reply:
I just verified that the RGW Pods are properly receiving their NodeAffinity. LGTM.